components/proxy/src/test/java/com/hotels/styx/proxy/backends/file/FileBackedBackendServicesRegistryTest.java

```diff
@@ -15,232 +15,142 @@
+        registry.addListener(listener);
+        registry.removeListener(listener);
 
-    @Test
-    public void firesChangeEventWhenBackendOriginIdsAreUpdated() throws IOException, ExecutionException, InterruptedException {
-        givenOrigins("/backends/origins-id-changed.yml");
-        ReloadResult result = reload().get();
-        assertThat(result.outcome(), is(RELOADED));
-        verify(changeListener).onChange(updated("webapp"));
+        verify(delegate).removeListener(eq(listener));
     }
 
     @Test
-    public void firesChangeEventWhenBackendOriginHostsAreUpdated() throws IOException, ExecutionException, InterruptedException {
-        givenOrigins("/backends/origins-host-changed.yml");
-        ReloadResult result = reload().get();
-        assertThat(result.outcome(), is(RELOADED));
-        verify(changeListener).onChange(updated("webapp"));
-    }
+    public void relaysGetToRgistryDelegate() {
+        FileBackedRegistry<BackendService> delegate = mock(FileBackedRegistry.class);
 
-    @Test
-    public void doesNotFireChangeEventWhenFileIsNotChanged() throws ExecutionException, InterruptedException {
-        ReloadResult result = reload().get();
-        assertThat(result.outcome(), is(UNCHANGED));
-        verifyNoMoreInteractions(changeListener);
-    }
+        registry = new FileBackedBackendServicesRegistry(delegate);
+        registry.get();
 
-    @Test
-    public void doesNotFireChangeEventWhenFileChangesDoNotChangeConfiguration() throws IOException, ExecutionException, InterruptedException {
-        givenOrigins("/backends/origins-comments-changed.yml");
-        ReloadResult result = reload().get();
-        assertThat(result.outcome(), is(UNCHANGED));
-        verifyNoMoreInteractions(changeListener);
+        verify(delegate).get();
     }
 
     @Test
-    public void firesChangeEventWhenBackendIsRemoved() throws IOException, ExecutionException, InterruptedException {
-        givenOrigins("/backends/origins-landing-removed.yml");
-        ReloadResult result = reload().get();
-        assertThat(result.outcome(), is(RELOADED));
-        verify(changeListener).onChange(removed("landing"));
+    public void reloadsDelegateRegistryOnStart() {
+        FileBackedRegistry<BackendService> delegate = mock(FileBackedRegistry.class);
+        when(delegate.reload()).thenReturn(completedFuture(reloaded("Changes applied!")));
+
+        registry = new FileBackedBackendServicesRegistry(delegate);
+        await(registry.start());
+
+        verify(delegate).reload();
     }
 
     @Test
-    public void firesChangeEventWhenBackendIsAdded() throws IOException, ExecutionException, InterruptedException {
-        givenOrigins("/backends/origins-pd-app-added.yml");
-        ReloadResult result = reload().get();
-        assertThat(result.outcome(), is(RELOADED));
-        verify(changeListener).onChange(added("product-details"));
-    }
+    public void passesThroughUnchangedReloadResult() {
```

Comment:
The name seems to suggest it's going to verify the output of the reload, but it doesn't. What is the objective here?

Reply:
I guess this can be removed.